src/core/mod.rs

```diff
@@ -105,25 +106,28 @@ pub struct BPFBuilder {
     cflags: Vec<CString>,
     device: Option<CString>,
     debug: BccDebug,
+    usdt_contexts: Vec<USDTContext>,
+    attach_usdt_ignore_pid: bool,
 }
 
 impl BPFBuilder {
     /// Create a new builder with the given code
     pub fn new(code: &str) -> Result<Self, BccError> {
-        let code = CString::new(code)?;
         Ok(Self {
-            code,
+            code: to_cstring(code)?,
```

Comment:
What's the motivation behind these changes?

Reply:
This could probably be removed.  I believe my thinking was that the error message crafted by `to_cstring` is better than the one `NulError` is coerced to -- which is just `null string` -- because like.. does that mean I'm using a "null" string?  Is "null" meaning "empty"? Have I given a null-terminated string? It was really just a "maybe I'm thinking too hard about this" thing. :)

Reply:
Thats fair. Unless there's a good way to handle that error conversion the motivation makes sense.

Reply:
Resolving this one. I think the better error message is a win